### ISSUE TITLE
Please sort list of model required fields

### ISSUE THREAD

**Issue by bgrant0607** [emicklei/go-restful#207]
Example where we get random reordering due to lack of deterministic sorting:
https://github.com/GoogleCloudPlatform/kubernetes/pull/8440/files

cc @nikhiljindal

**Comment by emicklei** [comment]
Do you want the fields to be lexicographically sorted or do they need to keep the order as listed in the structs? (i favor the latter because then the developer's effort to organize the list is not lost)

**Comment by bgrant0607** [comment]
The latter makes sense to me, also, but our primary requirement is determinism, so the only changes in output are deliberate.

**Comment by m0sth8** [comment]
I also want this to be fixed :+1:

**Comment by emicklei** [comment]
The issue207 branch has been merged into the master. Please verify expected behavior.

**Comment by bgrant0607** [comment]
Cool, thanks much. You just removed 3 files in one commit?
https://github.com/emicklei/go-restful/commit/bdfb7d41639a84ea7c36df648e5865cd9fbf21e2

@krousey Could you verify that this fixes sorting of required fields in our swagger output? If it does, then we can update our Godeps once this is merged into the master branch.

**Comment by emicklei** [comment]
@bgrant0607 , no it was the last commit of 5 commits to branch issue207, see github issue log.

**Comment by bgrant0607** [comment]
Ah, thanks. I didn't scroll down far enough. 
https://github.com/emicklei/go-restful/commits/issue207

**Comment by krousey** [comment]
@bgrant0607 this seems to work for us, I'll create a PR to pull it in.

**Comment by bgrant0607** [comment]
Thanks much.
